File: WebApplicationRefregitz/ThingsConverter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Drawing;
namespace RefrigtzW
{

    public class ThingsConverter
    {
        //Initiate Global Variables.
        bool ArrangmentsChanged = false;
        public static bool ActOfClickEqualTow = false;
        public bool Convert = false;
        public bool ConvertedToMinister = false;
        public bool ConvertedToBridge = false;
        public bool ConvertedToElefant = false;
        public bool ConvertedToHourse = false;
        public int Max;
        public int Row, Column;
        Color color;
        int Order;
        int Current = 0;
        public ThingsConverter()
        { }
        //Constructor
        public ThingsConverter(bool Arrangments, int i, int j, Color a, int[,] Tab, int Ord, bool TB, int Cur)
        {
            //Initite Global Variables with Local Parameter.
            ArrangmentsChanged = Arrangments;
            Row = i;
            Column = j;
            color = a;
            Order = Ord;
            Current = Cur;


        }
        //Convert Operation of Randomly All State Method.
        public bool ConvertOperation(int i, int j, Color a, int[,] Tab, int Ord, bool TB, int Cur)
        {
            //Initiate Global variables with Local One.
            Row = i;
            Column = j;
            color = a;
            Order = Ord;
            Current = Cur;
            //If Convert is Act and click tow time occured
            if (!Convert && ActOfClickEqualTow)
            {
                //Set tow time click unclicked.
                ActOfClickEqualTow = false;
                if (!ArrangmentsChanged)
                {
                    //Convert State Boolean Variable Consideration.
                    if (Order == 1 && Column == 7)
                        Convert = true;
                    if (Order == -1 && Column == 0)
                        Convert = true;
                }
                else
                {
                    //Convert State Boolean Variable Consideration.
                    if (Order == 1 && Column == 0)
                        Convert = true;
                    if (Order == -1 && Column == 7)
                        Convert = true;
                }
                //If Converted is Occured the Operation od Set and table reference content occured.
                if (Convert)
                {
                    if (!FormRefrigtz.Blitz)
                    {
                        AllDraw.SodierConversionOcuured = true;
                        //Randomly Number of 4 kind Object.
                        int Rand = (new Random()).Next(0, 4);
                        //If Rand is Equaled the Operation will cuased automaticcally Base on Color..
                        if (Rand == 0)
                        {
                            if (Order == 1)
                            {
                                // MinisterMidle++;
                                //MinisterHigh++;
                                Tab[Row, Column] = 5;
                            }
                            else if (Order == -1)
                            {
                                //MinisterHigh++;
                                Tab[Row, Column] = -5;
                            }
                            ConvertedToMinister = true;
                        }
                        else if (Rand == 1)
                        {
                            if (Order == 1)
                            {
                                //BridgeMidle++;
                                //BridgeHigh++;
                                Tab[Row, Column] = 4;
                            }
                            else if (Order == -1)
                            {
                                //BridgeHigh++;
                                Tab[Row, Column] = -4;
                            }
                            ConvertedToBridge = true;
                        }
                        else if (Rand == 2)
                        {
                            if (Order == 1)
                            {
                                //HourseMidle++;
                                //HourseHight++;
                                Tab[Row, Column] = 3;
                            }
                            else if (Order == -1)
                            {
                                //HourseHight++;
                                Tab[Row, Column] = -3;

                            }
                            ConvertedToHourse = true;
                        }
                        else
                        {
                            if (Order == 1)
                            {
                                //ElefantMidle++;
                                //ElefantHigh++;
                                Tab[Row, Column] = 2;
                            }
                            else if (Order == -1)
                            {
                                //ElefantHigh++;
                                Tab[Row, Column] = -2;
                            }
                            ConvertedToElefant = true;
                        }
                    }
                    else
                    {
                        if (FormRefrigtz.OrderPlate == 1)
                        {
                            AllDraw.SodierConversionOcuured = true;
                            //Randomly Number of 4 kind Object.
                            int Rand = (new Random()).Next(0, 4);
                            //If Rand is Equaled the Operation will cuased automaticcally Base on Color..
                            if (Rand == 0)
                            {
                                if (Order == 1)
                                {
                                    //  MinisterMidle++;
                                    //MinisterHigh++;
                                    Tab[Row, Column] = 5;
                                }
                                else if (Order == -1)
                                {
                                    //MinisterHigh++;
                                    Tab[Row, Column] = -5;
                                }
                                ConvertedToMinister = true;
                            }
                            else if (Rand == 1)
                            {
                                if (Order == 1)
                                {
                                    //BridgeMidle++;
                                    //BridgeHigh++;
                                    Tab[Row, Column] = 4;
                                }
                                else if (Order == -1)
                                {
                                    //BridgeHigh++;
                                    Tab[Row, Column] = -4;
                                }
                                ConvertedToBridge = true;
                            }
                            else if (Rand == 2)
                            {
                                if (Order == 1)
                                {
                                    //HourseMidle++;
                                    //HourseHight++;
                                    Tab[Row, Column] = 3;
                                }
                                else if (Order == -1)
                                {
                                    //HourseHight++;
                                    Tab[Row, Column] = -3;

                                }
                                ConvertedToHourse = true;
                            }
                            else
                            {
                                if (Order == 1)
                                {
                                    //ElefantMidle++;
                                    //ElefantHigh++;
                                    Tab[Row, Column] = 2;
                                }
                                else if (Order == -1)
                                {
                                    //ElefantHigh++;
                                    Tab[Row, Column] = -2;
                                }
                                ConvertedToElefant = true;
                            }
                        }


                        else
                        {



                            AllDraw.SodierConversionOcuured = true;
                            //Randomly Number of 4 kind Object.
                            int Rand = (new Random()).Next(0, 4);
                            //If Rand is Equaled the Operation will cuased automaticcally Base on Color..
                            if (Rand == 0)
                            {
                                if (Order == 1)
                                {
                                    //MinisterMidle++;
                                    //MinisterHigh++;
                                    Tab[Row, Column] = 5;
                                }
                                else if (Order == -1)
                                {
                                    //MinisterHigh++;
                                    Tab[Row, Column] = -5;
                                }
                                ConvertedToMinister = true;
                            }
                            else if (Rand == 1)
                            {
                                if (Order == 1)
                                {
                                    //BridgeMidle++;
                                    //BridgeHigh++;
                                    Tab[Row, Column] = 4;
                                }
                                else if (Order == -1)
                                {
                                    //BridgeHigh++;
                                    Tab[Row, Column] = -4;
                                }
                                ConvertedToBridge = true;
                            }
                            else if (Rand == 2)
                            {
                                if (Order == 1)
                                {
                                    //HourseMidle++;
                                    //HourseHight++;
                                    Tab[Row, Column] = 3;
                                }
                                else if (Order == -1)
                                {
                                    //HourseHight++;
                                    Tab[Row, Column] = -3;

                                }
                                ConvertedToHourse = true;
                            }
                            else
                            {
                                if (Order == 1)
                                {
                                    //ElefantMidle++;
                                    //ElefantHigh++;
                                    Tab[Row, Column] = 2;
                                }
                                else if (Order == -1)
                                {
                                    //ElefantHigh++;
                                    Tab[Row, Column] = -2;
                                }
                                ConvertedToElefant = true;
                            }
                        }
                    }
                    return Convert;
                }
            }
            //return Convert State.
            return Convert;

        }
    }
}
//End of Documeatation.

File: Refrigtz/ThingsConverter.vb
Imports System.Drawing
Imports System.Text
Imports System.Linq
Imports System.Collections.Generic
Imports System

Namespace Refrigtz

	Public Class ThingsConverter
		'Initiate Global Variables.
		Public Shared ActOfClickEqualTow As Boolean = False
		Public Convert As Boolean = False
		Public ConvertedToMinister As Boolean = False
		Public ConvertedToBridge As Boolean = False
		Public ConvertedToElefant As Boolean = False
		Public ConvertedToHourse As Boolean = False
		Public Max As Integer
		Public Row As Integer, Column As Integer
		Private color As Color
		Private Order As Integer
		Private Current As Integer = 0
		Public Sub New()
		End Sub
		'Constructor
		Public Sub New(i As Integer, j As Integer, a As Color, Tab As Integer(,), Ord As Integer, TB As Boolean, _
			Cur As Integer)
			'Initite Global Variables with Local Parameter.
			Row = i
			Column = j
			color = a
			Order = Ord


			Current = Cur
		End Sub
		'Convert Operation of Randomly All State Method.
		Public Function ConvertOperation(i As Integer, j As Integer, a As Color, Tab As Integer(,), Ord As Integer, TB As Boolean, _
			Cur As Integer) As Boolean
			'Initiate Global variables with Local One.
			Row = i
			Column = j
			color = a
			Order = Ord
			Current = Cur
			'If Convert is Act and click tow time occured
			If Not Convert AndAlso ActOfClickEqualTow Then
				'Set tow time click unclicked.
				ActOfClickEqualTow = False
				'Convert State Boolean Variable Consideration.
				If color = Color.Gray AndAlso Column = 7 Then
					Convert = True
				End If
				If color = Color.Brown AndAlso Column = 0 Then
					Convert = True
				End If
				'If Converted is Occured the Operation od Set and table reference content occured.
				If Convert Then

					AllDraw.SodierConversionOcuured = True
					'Randomly Number of 4 kind Object.
					Dim Rand As Integer = (New Random()).[Next](0, 4)
					'If Rand is Equaled the Operation will cuased automaticcally Base on Color..
					If Rand = 0 Then
						If color = Color.Gray Then
							AllDraw.MinisterMidle += 1
							AllDraw.MinisterHigh += 1
							Tab(Row, Column) = 5
						ElseIf color = Color.Brown Then
							AllDraw.MinisterHigh += 1
							Tab(Row, Column) = -5
						End If
						ConvertedToMinister = True
					ElseIf Rand = 1 Then
						If color = Color.Gray Then
							AllDraw.BridgeMidle += 1
							AllDraw.BridgeHigh += 1
							Tab(Row, Column) = 4
						ElseIf color = Color.Brown Then
							AllDraw.BridgeHigh += 1
							Tab(Row, Column) = -4
						End If
						ConvertedToBridge = True
					ElseIf Rand = 2 Then
						If color = Color.Gray Then
							AllDraw.HourseMidle += 1
							AllDraw.HourseHight += 1
							Tab(Row, Column) = 3
						ElseIf color = Color.Brown Then
							AllDraw.HourseHight += 1

							Tab(Row, Column) = -3
						End If
						ConvertedToHourse = True
					Else
						If color = Color.Gray Then
							AllDraw.ElefantMidle += 1
							AllDraw.ElefantHigh += 1
							Tab(Row, Column) = 2
						ElseIf color = Color.Brown Then
							AllDraw.ElefantHigh += 1
							Tab(Row, Column) = -2
						End If
						ConvertedToElefant = True
					End If
					Return Convert
				End If
			End If
			'return Convert State.
			Return Convert

		End Function
	End Class
End Namespace